ui/apps/moderation/src/extensions/report-modal.tsx

```diff
@@ -49,10 +45,6 @@ const ReportModalComponent = (props: RootExtensionProps) => {
 
   const onReport = React.useCallback(
     (dataToSign: Record<string, string>) => {
-      analyticsActions.trackEvent({
```

Comment:
Why did you moved the analytics into the hook?

Reply:
So it tracks, only if the report action was successfully executed. On the modal method, it will send the action regardless of if the mutation was successful or not

Reply:
1. If is successful then you will see the entry in the moderation app so this event becomes redundant.
2. This event should be recorded regardless of the server's response... (aka The reported is willing to do this action. If there is an error, that's another story)
3. If you really want to do it on success, you can do it in an effect hook without mixing things out.